nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/Settings.java

```diff
@@ -78,9 +82,14 @@ public Settings(int defaultConcurrency, int defaultPartitions, Codec<Record> def
         this(defaultConcurrency, defaultPartitions, defaultCodec, defaultPolicy, null);
     }
 
-    @SuppressWarnings("unchecked")
     public Settings(int defaultConcurrency, int defaultPartitions, Codec<Record> defaultCodec,
             ComputationPolicy defaultPolicy, RecordFilterChain defaultFilter) {
+        this(defaultConcurrency, defaultPartitions, defaultCodec, defaultPolicy, defaultFilter, false);
+    }
+
+    @SuppressWarnings("unchecked")
+    public Settings(int defaultConcurrency, int defaultPartitions, Codec<Record> defaultCodec,
+            ComputationPolicy defaultPolicy, RecordFilterChain defaultFilter, boolean defaultExternal) {
```

Comment:
This is getting messy, you probably want to use a builder pattern?

Reply:
yes I have created NXP-28362 for this

Reply:
with a new commit